_posts/2020-04-24-roboarmcontroller-intro.md

```diff
@@ -0,0 +1,197 @@
+   whereas the right hand (yellow arrow) indicates a positive or negative delta</figcaption>
+</figure>
+
+This fits the learning goals (mentioned on the afterword) because:
+1. The RoboArm Controller is a data stream, where the data originates on the sensor, is transformed, and eventually it "moves"
+the simulator. 
+1. The RoboArm Controller can be controlled with a machine learning part because the transformation of the data is a classification problem, 
+where the positions of the hand have a label, which itself can be mapped to an instruction. 
+1. The RoboArm Controller enables the practice of the machine learning framework because it is an end to end machine learning system,
+ executing the following steps: Gather the data, prepare the data, choose a model, train the model, tune the hyperparameters,
+ test the model, export the model, import the model into a "production" system, and use it to make predictions. 
+
+
+## Architecture
+
+This is a data stream: the data is originated on the hands' sensor, then it is read by the data transformation
+application, where it is transformed from a set of positions to either a data set or to a movement instruction, and on the latter case sent
+to the robotic arm, where the instruction is interpreted and applied to the simulation.
+
+It has 3 main components:
+1. A hands sensor.
+1. A data transformer application.
+1. A simulated robotic arm.
+
+Note: I don't consider the Machine Learning module as a component, from my perspective it is more a tool used by a component or even an implementation detail.
+I show it here because it is a goal on the project: learn a machine learning framework.
+
+<figure>
+  <img src="/assets/posts/roboarmcontroller-intro/target-architecture.png" alt="The current architecture"/>
+  <figcaption class="image-description">The current architecture of the RoboArm Controller.</figcaption>
+</figure>
+
+If you want to see more diagrams related to the architecture of this project, you can find them at the
+ [Architecture folder of the GitHub repo][roboarmcontroller_repo_architecture]
+
+## Brief description of the components
+
+This is just an overview of each of the main components used for the project. If these series get enough traction, I will
+write a post describing each one of them
+
+### Hands' sensor
+
+This is the origin of the data, where the position of the hands is sensed.
+Used the [Leap Motion][leap_motion] sensor. It has a framework to communicate with different languages, but the way I got to talk 
+with it was through a WebSocket. It provides coordinates for a set of points related to the hands, e.g. the tips of the fingers,
+the center of the hand, and so on.
+- Technologies used: Leap Motion.
+- Interfaces: Websocket.
+- Input: The hands.
+- Output: Position of different parts of the hands, sent to the data transformer application.
```

Comment:
As it's a classification problem, the output should be a rather simple data model? E.g. a 3-dimensional vector with an entry for each servo?  Maybe that is quicker/easier to understand than the text?

Reply:
this comment applies to the output of the data transformer application. Added an example of the output there